src/main/java/de/dennisguse/opentracks/settings/PreferencesUtils.java

```diff
@@ -576,46 +574,80 @@ public static void setLayoutColumns(int columns) {
     }
 
     @SuppressLint("ResourceType")
-    static String buildDefaultLayout() {
+    private static String buildDefaultFields() {
         List<TypedArray> fieldsArrays = getMultiTypedArray("stats_custom_layout_fields_default_value");
-        return resources.getString(R.string.default_activity_default) + CsvConstants.LINE_SEPARATOR
-                + fieldsArrays.stream().map(i -> i.getString(0) + CsvConstants.ITEM_SEPARATOR + i.getString(1)).collect(Collectors.joining(CsvConstants.LINE_SEPARATOR))
-                + CsvConstants.LINE_SEPARATOR;
+        return fieldsArrays.stream().map(i -> i.getString(0) + CsvConstants.PROPERTY_SEPARATOR + i.getString(1)).collect(Collectors.joining(CsvConstants.ITEM_SEPARATOR))
+                + CsvConstants.ITEM_SEPARATOR;
+    }
+
+    static String buildDefaultLayout() {
+        return resources.getString(R.string.stats_custom_layout_default_profile) + CsvConstants.ITEM_SEPARATOR + getLayoutColumnsByDefault() + CsvConstants.ITEM_SEPARATOR + buildDefaultFields();
     }
 
+    /**
+     * @return custom layout selected or the first one if any has been selected or the one selected is not exists anymore.
+     */
     public static Layout getCustomLayout() {
-        String csvCustomLayout = getString(R.string.stats_custom_layout_fields_key, buildDefaultLayout());
-        List<String> csvParts = Arrays.asList(csvCustomLayout.split(CsvConstants.LINE_SEPARATOR));
-        Layout layout = new Layout(csvParts.get(0));
-        for (int i = 1; i < csvParts.size(); i++) {
-            String[] fieldParts = csvParts.get(i).split(CsvConstants.ITEM_SEPARATOR);
-            layout.addField(fieldParts[0], DataField.getTitleByKey(resources, fieldParts[0]), fieldParts[1].equals(DataField.YES_VALUE), fieldParts[2].equals(DataField.YES_VALUE), fieldParts[0].equals(resources.getString(R.string.stats_custom_layout_coordinates_key)));
+        String csvCustomLayouts = getString(R.string.stats_custom_layouts_key, buildDefaultLayout());
+        String[] csvLines = csvCustomLayouts.split(CsvConstants.LINE_SEPARATOR);
+        String profileSelected = getString(R.string.stats_custom_layout_selected_profile_key, null);
+        if (profileSelected == null) {
+            return Layout.fromCsv(csvLines[0], resources);
```

Comment:
csvLines[0] might not exist; why not just create a Layout instead of parse the CSV?